core/domain/stats_jobs_one_off.py

```diff
@@ -397,19 +400,30 @@ def prepare_reduce(cls, exp_id, values):
                 exp_stats_dict = copy.deepcopy(prev_stats_dict)
 
             # Compute the statistics for events corresponding to this version.
+            exp_started_session_ids, exp_completed_session_ids = set(), set()
+            exp_actually_started_session_ids = set()
             state_hit_session_ids, solution_hit_session_ids = set(), set()
             while event_dict['version'] == version:
                 state_stats = (exp_stats_dict['state_stats_mapping'][
                     event_dict['state_name']])
                 if event_dict['event_type'] == (
                         feconf.EVENT_TYPE_START_EXPLORATION):
-                    exp_stats_dict['num_starts_v2'] += 1
+                    if event_dict['session_id'] not in exp_started_session_ids:
+                        exp_stats_dict['num_starts_v2'] += 1
+                    exp_started_session_ids.add(event_dict['session_id'])
```

Comment:
Here and in the next two stanzas, you might as well move this line under the "if", right?